src/main/java/fi/aalto/cs/apluscourses/ui/utils/TreeModelTraversal.java

```diff
@@ -0,0 +1,60 @@
+package fi.aalto.cs.apluscourses.ui.utils;
+
+import fi.aalto.cs.apluscourses.utils.Streamable;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+import javax.swing.tree.TreeModel;
+import javax.swing.tree.TreePath;
+import org.jetbrains.annotations.NotNull;
+
+public class TreeModelTraversal {
```

Comment:
Is there a reason that this isn't a utility class with `TreeModel` as a method parameter instead of an instance variable?

Reply:
Well, I'd say this is more OOP way to do it - no need to pass the same object reference first to `traverse(TreePath)` then to `getChildNodeStream(Object)`, then to `NodeIterator`. Is there a reason to do it in other way?

The name of the class is, however, a bit misleading. Perhaps `TreeModelTraverser` would be better. After all, the same object can be used for more than one traversals.

Reply:
It's not used in `getChildNodeStream` which is why I initially asked, it seemed like an overly complicated way of passing a parameter. Now I see that the iterator uses it, so this way makes sense. I agree that `Traverser` is a better name :+1: